### PR TITLE
[KM181] Manage the lifecycle of RDS Postgres (Part 2)

### PR DESCRIPTION
**Pull request by paulbes**
See #387 for more details.

**NB:** Do not merge until #390 has been merged into this PR

## Description
In this PR we implement the server and client sides and route up most of the things. We also create a hidden command for testing the setup. There will come more PRs, which will contain the reconciler for declaring the Postgres databases, a way of attaching to the database and a lambda for the rotation schedule. 

Preview of outputs:

```yaml
databases:
    mydb:
        type: postgres
        namespace: mydb
        endpointaddress: op1e2zk1kiflbdb.c5oiqv7tmmdx.eu-west-1.rds.amazonaws.com
        endpointport: 5432
        securitygroupid: sg-0a020b6f07f0a5c0e
        adminsecretname: mydb-postgres-admin
        databaseconfigmapname: mydb-postgres
```

```bash
$ kubectl describe cm mydb-postgres -n mydb
Name:         mydb-postgres
Namespace:    mydb
Labels:       <none>
Annotations:  <none>
Data
====
endpointAddress:
----
op1e2zk1kiflbdb.c5oiqv7tmmdx.eu-west-1.rds.amazonaws.com
endpointPort:
----
5432
Events:  <none>
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated the release notes (for the next release).

### CODE REVIEW

File: pkg/okctl/okctl.go [hunk 1]
```diff
@@ -30,8 +30,6 @@ import (
 
 	"github.com/oslokommune/okctl/pkg/config/state"
 
-	"github.com/oslokommune/okctl/pkg/api/core/store/noop"
```
Comment (ivaruf) [inline pkg/okctl/okctl.go:33]: What were we using noop for before, and what replaced them anyway?

**Review by ivaruf — APPROVED** [review]

File: pkg/okctl/okctl.go (hunk 1 above)
Comment (paulbes) [reply pkg/okctl/okctl.go:33]: It was introduced when we were transitioning from storing things on the filesystem serverside, just a way of not having to refactor _everything_ at once

File: pkg/cfn/components/composer.go [hunk 2]
```diff
@@ -290,14 +288,24 @@ func (e *ExternalSecretsPolicyComposer) ManagedPolicy() *managedpolicy.ManagedPo
...
 				Action: []string{
-					"secretsmanager:GetSecretValue",
 					"ssm:GetParameter",
 				},
 				Resource: []string{
-					asmParameterARN("*"),
 					ssmParameterARN("*"),
 				},
 			},
+			{
```
Comment (deifyed) [inline pkg/cfn/components/composer.go:297]: My bad. Thanks

**Review by deifyed — APPROVED** [review]

File: pkg/api/core/cloudprovider/aws/component_aws.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/oslokommune/okctl/pkg/cfn"
+	"github.com/oslokommune/okctl/pkg/cfn/components"
+)
+
+type componentCloudProvider struct {
+	provider v1alpha1.CloudProvider
+}
+
+const (
+	postgresTimeOutInMinutes = 45
```
[the remark this answers is not in the document]
Comment (paulbes) [reply pkg/api/core/cloudprovider/aws/component_aws.go:17]: > Stopping and starting a DB will make AWS automatically create a snapshot (I was told during the AWS Sys admin course). Creating snapshots is very fast usually, but sometimes there are very large delays. My personal view is that this is probably some form of congestion on AWS' side, but I don't have an official comment to verify this.

When I have been testing this comment from [stack overflow](https://stackoverflow.com/questions/60058911/why-does-an-rds-instance-take-2-hours-to-start) seems to be the most accurate reason

File: pkg/cfn/testdata/vpc-cloudformation.yaml.golden [hunk 4]
```diff
@@ -6,6 +6,16 @@ Outputs:
...
     Value:
       Ref: DatabaseSubnetGroup
+  DatabaseSubnetIds:
+    Export:
+      Name:
+        Fn::Sub: ${AWS::StackName}-DatabaseSubnetIds
+    Value:
+      Fn::Join:
+      - ','
+      - - Ref: DatabaseSubnet00
```
[the remark this answers is not in the document]
Comment (paulbes) [reply pkg/cfn/testdata/vpc-cloudformation.yaml.golden:16]: There is one database subnet per [availability zone](https://docs.aws.amazon.com/AmazonElastiCache/latest/mem-ug/RegionsAndAZs.html), and in `eu-west-1` there are 3 availability zones, good question!

**Pull request merged**